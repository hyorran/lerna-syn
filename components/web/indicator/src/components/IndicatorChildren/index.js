import React from 'react'
import PropTypes from 'prop-types'
import { withStyles } from '@material-ui/core/styles'
import Grow from '@material-ui/core/Grow'
import Indicator from '../Indicator'

import styles from './styles'

/**
 * @return {null}
 */
function IndicatorChildren(props) {
  const {
    checked,
    onClick,
    indicatorProps,
    timeout
  } = props

  if (!checked) {
    return null
  }

  return (
    <Grow
      in={ checked }
      timeout={ timeout }
    >
      <div
        onClick={ onClick }
        role="presentation"
        onKeyPress={ onClick }
      >
        <Indicator
          inGroup
          manualRefresh={ false }
          { ...indicatorProps }
        />
      </div>
    </Grow>
  )
}

IndicatorChildren.propTypes = {
  checked: PropTypes.bool,
  onClick: PropTypes.func,
  indicatorProps: PropTypes.object,
  timeout: PropTypes.number
}

IndicatorChildren.defaultProps = {
  checked: false,
  onClick: () => {},
  indicatorProps: {},
  timeout: 0
}

export default withStyles(styles)(IndicatorChildren)
