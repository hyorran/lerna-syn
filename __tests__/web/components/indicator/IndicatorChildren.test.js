import React from 'react'
import IndicatorChildren from '@syntesis/c-indicator/src/components/IndicatorChildren'
import { mount } from 'enzyme'

describe('<IndicatorChildren />', () => {
  it('rendered', () => {
    const wrapper = mount((
      <IndicatorChildren />
    ))
    expect(wrapper).toMatchSnapshot()
  })
})
