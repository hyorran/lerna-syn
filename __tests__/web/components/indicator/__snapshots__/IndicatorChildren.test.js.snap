// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IndicatorChildren /> rendered 1`] = `
<WithStyles(IndicatorChildren)>
  <IndicatorChildren
    checked={false}
    classes={Object {}}
    indicatorProps={Object {}}
    onClick={[Function]}
    timeout={0}
  />
</WithStyles(IndicatorChildren)>
`;
